### PR TITLE
Add modified_beam_search for streaming decode

### PR DESCRIPTION
**Pull request by pkufool**
- [ ] Add to other recipes

### CODE REVIEW

File: egs/librispeech/ASR/pruned_transducer_stateless/streaming_beam_search.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    context_size = model.decoder.context_size
+    device = model.device
+    T = encoder_out.size(1)
+
+    decoder_input = torch.tensor(
+        [stream.hyp[-context_size:] for stream in streams],
+        device=device,
+        dtype=torch.int64,
+    )
+    # decoder_out is of shape (N, decoder_out_dim)
```
Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless/streaming_beam_search.py:58]:
```suggestion
    # decoder_out is of shape (N, 1, decoder_out_dim)
```

File: egs/librispeech/ASR/pruned_transducer_stateless/streaming_beam_search.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+                decoder_input,
+                need_pad=False,
+            )
+
+
+def modified_beam_search(
+    model: nn.Module,
+    encoder_out: torch.Tensor,
+    streams: List[DecodeStream],
+    beam: int = 4,
```
Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless/streaming_beam_search.py:96]: I suggest using `num_active_paths` in the **new** code.

People can get confused with `beam` and `beam_size`.

File: egs/librispeech/ASR/pruned_transducer_stateless/streaming_decode.py [hunk 3]
```diff
@@ -31,7 +31,6 @@
 import argparse
 import logging
 import math
-import warnings
```
Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless/streaming_decode.py]: In line 20, it should be
```
./pruned_transducer_stateless/streaming_decode.py \
```

File: egs/librispeech/ASR/pruned_transducer_stateless/streaming_beam_search.py (hunk 2 above)
Comment (pkufool) [reply egs/librispeech/ASR/pruned_transducer_stateless/streaming_beam_search.py:96]: I know you renamed it to `num_active_paths` in sherpa, I thought it is better to make it be consistent with the one in `decode.py`. So do we need to rename the one in `decode.py`?

Comment (csukuangfj) [reply egs/librispeech/ASR/pruned_transducer_stateless/streaming_beam_search.py:96]: I suggest only doing this for **new** code.

The current file associates two different meanings with `beam`, which can confuse users, I think.
https://github.com/k2-fsa/icefall/blob/72d76a4ff8aa43b88f967a37c0346590e4ecbfca/egs/librispeech/ASR/pruned_transducer_stateless/streaming_beam_search.py#L96

https://github.com/k2-fsa/icefall/blob/72d76a4ff8aa43b88f967a37c0346590e4ecbfca/egs/librispeech/ASR/pruned_transducer_stateless/streaming_beam_search.py#L203

**Pull request merged**